test/comparison/testdata/fbs/property-update/c/ChangeUid.obx.h.expected

```diff
@@ -109,3 +108,71 @@ static void ChangeUid_free(ChangeUid** object) {
     *object = NULL;
 }
 
+/// Insert or update the given object in the database.
+/// @param object (in & out) will be updated with a newly inserted ID if the one specified previously was zero. If an ID 
+/// was already specified (non-zero), it will remain unchanged.
+/// @return object ID from the object param (see object param docs) or a zero on error. If a zero was returned, you can
+/// check obx_last_error_*() to get the error details. In an unlikely event that those functions return no error
+/// code/message, the error occurred in FlatBuffers serialization, e.g. due to memory allocation issues.
+static obx_id ChangeUid_put(OBX_store* store, ChangeUid* object) {
+    flatcc_builder_t builder;
+    flatcc_builder_init(&builder);
+
+    obx_id id = 0;
+    size_t size = 0;
+    void* buffer = NULL;
+    if (!ChangeUid_to_flatbuffer(&builder, object, &buffer, &size)) {
+        // if the FlatBuffers serialization fails, we clear any previous ObjectBox errors, so that when our caller
```

Comment:
just read the end of the sentence ;D 

if we don't clear the error, the caller may get an invalid (completely unrelated) error when checking `obx_last_error_code()`

Reply:
Hmm, please suggest a change to the method comment, I thought it's pretty thorough explaining how to do error checking:

current version:
```
	/// @return object ID from the object param (see object param docs) or a zero on error. If a zero was returned, you can
/// check obx_last_error_*() to get the error details. In an unlikely event that those functions return no error
/// code/message, the error occurred in FlatBuffers serialization, e.g. due to memory allocation issues.
```

WRT to clearing, I'm not aware of any other place this situation occurs. The alternative I was considering was being able to call sth like `obx_last_error_set()` but since that's not available, this was the only other approach (with deterministic results) I came up with.

Reply:
Probably just allocation but I'm not so familiar with flatcc to say there can't fail for any other reason.... 


I've tried to go ahead with returning the error code until I've realized it's still the same problem. See the following user code:
```c
obx_err err = foo_put();
if (err != OBX_SUCCESS) {
    printf("Last error: %s (%d)\n", obx_last_error_message(), code);
}
```
So if the foo_put() returns an error code on FB serialization failure, the code prints no error text or an invalid one (if any occurred previously). 

I can think of the following options:
1. introduce yet another layer of error handling - in the generated code (not really an option IMO)
1. what I've already done above, i.e. clearing the error
1. add `obx_last_error_set(obx_err, const char*)` and call it from the generated code when an error occurs in FB allocation/building

Reply:
changed to error_set